### ISSUE TITLE
Question about using openpgp in a commercial electron app

### ISSUE THREAD

**Issue by jano-m** [openpgpjs/openpgpjs#645]
We would like to use npm package openpgp in our electron application. Our understanding of LGPL license is that in order to satisfy licensing requirements, we either need to release the source code of our application (which we would not like to do) or link openpgp dynamically during run-time. How can we do the later? It does not seem that building electron app links packages dynamically. Could you give us an advice on this? Thanks.

**Comment by jano-m** [comment]
We are using electron-builder: https://github.com/electron-userland/electron-builder

**Comment by bartbutler** [comment]
You could try using https://www.electron.build/configuration/contents#extrafiles to get openpgp.min.js into the app content directory and then use a relative path require to load it. My (very limited) understanding of electron builder is that it still relies on webpack or similar to bundle it's files, right? So you'd have to exclude the require there.

**Comment by jano-m** [comment]
Thanks! I think we achieved the setup acceptable by the license. I will describe it here, perhaps it might help other people using electron-builder.

In package.json: 
  - We removed "openpgp" from "dependencies" and added it to "devDependencies". This way it's not going to be bundled in app.asar.
  - In "build" section we added (as you suggested)
          `"extraFiles": "node_modules/openpgp/dist/openpgp.min.js",`

It was a bit tricky to load the package in our app, however. It's not enough to use require('openpgp') since it's not in app.asar, but in the subfolder of the installation directory, which can be different on different systems. We used app.getPath('exe') to obtain the path the main executable file of our app and then use it to find openpgp.min.js file:
`const {app} = require('electron');`
`const path = require('path');`
`const openpgp = require(path.join(app.getPath('exe'),`
`    (process.platform === 'darwin' ? '../' : '')+'../node_modules/openpgp/dist/openpgp.min.js'));`

A bit awkward, but it works. If anyone has a better suggestion how to load openpgp.min.js, we would like to hear it.

Also, let us know if you think there is still problem with the license with this way of loading the openpgp module.
